app/models/miq_report.rb

```diff
@@ -89,6 +90,22 @@ def self.having_report_results(options = {})
     q
   end
 
+  def col_format_with_defaults
+    return [] unless cols.present?
+
+    cols.each_with_index.map do |column, index|
+      column_format = col_formats.try(:[], index)
+      if column_format
+        column_format
+      else
+        column = Chargeback.default_column_for_format(column.to_s) if Chargeback.db_is_chargeback?(db)
```

Comment:
wish we didn't need this here

Reply:
I am working on some extraction code from` MiqReport#format `and it will allow us to do something with it.